modules/PacketSniffer/PacketSniffer.cpp

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Pionix GmbH and Contributors to EVerest
+#include "PacketSniffer.hpp"
+
+#include <fmt/core.h>
+
+namespace module {
+
+const bool PROMISC_MODE = true;
+const int PACKET_BUFFER_TIMEOUT_MS = 1000;
+
+void PacketSniffer::init() {
+    invoke_init(*p_main);
+
+    capturing_stopped = false;
+
+    handle = pcap_open_live(config.device.c_str(), BUFSIZ, PROMISC_MODE, PACKET_BUFFER_TIMEOUT_MS, errbuf);
+    if (handle == NULL) {
+        EVLOG_AND_THROW(Everest::EverestConfigError(fmt::format("Could not open device {}", config.device)));
+        return;
+    }
+
+    if (pcap_datalink(handle) != DLT_EN10MB) {
+        EVLOG_AND_THROW(Everest::EverestConfigError(fmt::format("Device {} doesn't provide Ethernet headers - not supported", config.device)));
+        return;
+    }
+
+    r_evse_manager->subscribe_session_event([this](types::evse_manager::SessionEvent session_event) {
+        auto event = types::evse_manager::session_event_enum_to_string(session_event.event);
+        if (event == "SessionStarted") {
+            capturing_stopped = false;
+            std::thread(&PacketSniffer::capture, this, config.session_logging_path, session_event.uuid).detach();
+        } else if (event == "SessionFinished") {
+            capturing_stopped = true;
+        }
+    });
+}
```

Comment:
What if multiple SessionStarted without any SessionFinished happen? Then it is tried multiple times to attach via pcap.  I would add a mutex to this session event listener and check first if a session is already running

Reply:
Good Point!

Reply:
Should now work like this